.github/workflows/build.yml

```diff
@@ -40,6 +40,10 @@ jobs:
         if: inputs.build_from_code_artifact != 'true'
         uses: actions/checkout@v4
 
+      - name: set environment variables
+        run: |
+          echo "TODAY=$(date +%Y-%m-%d)" >> $GITHUB_ENV
```

Comment:
Can you tell me why we need a today variable?

Reply:
I guess for the cache key later down, but - still I feel like I've looked at many actions that didn't use this that I'd like to discuss it more generally. Maybe in the comp meeting today?

Reply:
This is how both the [setup-micromamba](https://github.com/mamba-org/setup-micromamba) and [setup-miniconda](https://github.com/conda-incubator/setup-miniconda) examples demonstrate it.  

I have seen some projects with a cache cleanup GitHub action (e.g. Pandas) instead, which might be a good idea, but would be more verbose not less 🙃 .

Reply:
Yeah! We can also discuss this in the comp meeting later, but the tl;dr is what Katelyn said. Adding the date to the cache key makes sure the environment remakes at least once every 24 hours.